src/porting.cpp

```diff
@@ -39,6 +39,7 @@
 #endif
 #if defined(__APPLE__)
 	#include <mach-o/dyld.h>
+	#import <Foundation/Foundation.h>
```

Comment:
This is an Objective-C/C++ framework. It cannot be used in C++ code directly. You have to do an Objective-C++ wrapper, I assume.
Something like porting_ios files in #15451